### PR TITLE
Allow filtering by target branch

### PR DESCRIPTION
**Pull request by marionebl**
This introduces a feature for `mr list`, where users can filter the list of printed merge requests by target branch.

### CODE REVIEW

**Comment by marionebl** [comment]
A hint about differences between local and remote testing would be greatly appreciated :)

File: cmd/mr_list_test.go [hunk 1]
```diff
@@ -88,3 +88,33 @@ func Test_mrListFivePerPage(t *testing.T) {
...
+	cmd := exec.Command("../lab_bin", "mr", "list", "-t", "non-existing")
+	cmd.Dir = repo
+
+	b, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mrs := strings.Split(string(b), "\n")
+	require.Equal(t, "PASS", mrs[0])
```
Comment (zaquestion) [inline cmd/mr_list_test.go]: "PASS" will always be present in the output (assuming non-zero exec which will show as an err to the exec call). Its better to assert that the length only has the test prints in it. I think there are 2? lines. so something like `require.Equal(t, 2, len(mrs))

**Comment by zaquestion** [comment]
@marionebl thanks for contributing. It looks like the GitLab API supports filtering by branch name https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests

You should just be able to specify `TargetBranch` in the just above where the MRs are listed `gitlab.ListProjectMergeRequestsOptions`

Testing wise you should just need an ssh key attached to a gitlab account for `make test` to work and you can use `make test run=<regex>` (same as you'd use with go test -run=). The data its looking at lives in https://gitlab.com/zaquestion/test/merge_requests

**Comment by marionebl** [comment]
> @marionebl thanks for contributing. It looks like the GitLab API supports filtering by branch name https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests

> You should just be able to specify TargetBranch in the just above where the MRs are listed gitlab.ListProjectMergeRequestsOptions

Being new to Go as an ecosystem I originally wanted to avoid the go-gitlab update required to do this, gave it a try now. 

It appears to break some other functionality, though: https://travis-ci.org/zaquestion/lab/builds/433313336#L567

I adapted the test case as suggested.

**Comment by marionebl** [comment]
Fixed the issue with `ApprovalsBeforeMerge` by defaulting it to `0`

**Pull request merged**